src/Microsoft.AspNet.Authentication/AuthenticationHandler.cs

```diff
@@ -61,12 +64,13 @@ internal AuthenticationOptions BaseOptions
 
         public bool Faulted { get; set; }
 
-        protected async Task BaseInitializeAsync([NotNull] AuthenticationOptions options, [NotNull] HttpContext context, [NotNull] ILogger logger)
+        protected async Task BaseInitializeAsync([NotNull] AuthenticationOptions options, [NotNull] HttpContext context, [NotNull] ILogger logger, [NotNull] IUrlEncoder encoder)
         {
             _baseOptions = options;
             Context = context;
             RequestPathBase = Request.PathBase;
             Logger = logger;
+            UrlEncoder = encoder;
```

Comment:
I only count 3 handlers that use it, cookies, facebook, and OIDC. That's probably insufficient for putting it in the base class.